### PR TITLE
Avoid confusion if multiple Visual Studios are installed

### PR DESCRIPTION
**Pull request by dzenanz**

### CODE REVIEW

File: superbuild/External-Boost.cmake [hunk 1]
```diff
@@ -29,13 +29,25 @@ else()
...
+endif(MSVC_VERSION EQUAL 1500)
+
 ExternalProject_Add(Boost
   BUILD_IN_SOURCE 1
   URL ${Boost_url}
   URL_MD5 ${Boost_md5}
   UPDATE_COMMAND ""
   CONFIGURE_COMMAND ${Boost_Bootstrap_Command} --prefix=${INSTALL_DEPENDENCIES_DIR}/lib
-  BUILD_COMMAND ${Boost_b2_Command} install -j8   --prefix=${INSTALL_DEPENDENCIES_DIR} --with-thread --with-filesystem --with-system --with-date_time --with-program_options address-model=${Boost_address_model}
+  BUILD_COMMAND ${Boost_b2_Command} install -j8   --prefix=${INSTALL_DEPENDENCIES_DIR} --with-thread --with-filesystem --with-system --with-date_time --with-program_options address-model=${Boost_address_model} --toolset=${boost_toolset}
```
Comment (arnaudgelas) [inline superbuild/External-Boost.cmake]: When someone does not use MSVC, the the BUILD_COMMAND will end with '--toolset=' ??
wouldn't it be better to store "--toolset=" into one variable only set for provided MSVC_VERSION ?

Comment (dzenanz) [inline superbuild/External-Boost.cmake]: You are completely right. I will change that now.

**Comment by dzenanz** [comment]
I changed the code and force-pushed it to my fork. Do I need to do something more?

**Comment by arnaudgelas** [comment]
Let's wait for AppVeyor and travis to report compilation results (which should be ok), then you can open another PR for the develop branch

**Comment by tom-albrecht** [comment]
Thanks for looking into this. I just ran into the same issue.

**Comment by tom-albrecht** [comment]
Can you explain why this is needed? 
This seems like we would need to add a new line for every Visual Studio version that comes out in the future?

**Comment by dzenanz** [comment]
By default, boost decided to build itself with VS2015 even though I was running it from VS2013 command prompt. The same thing happens in superbuild: I build Statismo with VS2013, but boost gets built by VS2015 and produces tons for link errors. Since boost is not using the standard way of deciding which compiler to be built with (check which cl.exe is in path), we need to specify "toolset". Unless boost changes its build behavior, we will need to extend that if/else statement every time a new VS comes out (approximately ever 1-2 years).

**Comment by arnaudgelas** [comment]
I don't think appveyor is well connected to statismo. Can anyone confirm it actually works?

**Comment by dzenanz** [comment]
The reason to not build everything in VS2015 is that not all libraries (in versions which Statismo uses) can be built with VS2015 - there are a lot of errors even during CMake configuration step (applications which CMake tries to compile and execute crash).

**Comment by tom-albrecht** [comment]
OK. That makes a lot of sense. That's why I ran into the same issue. 
I just installed VS2015 but still use VS2013 for statismo.

**Comment by tom-albrecht** [comment]
FYI: I am trying it out, but the boost builds take forever. I can probably only report next week.

**Comment by dzenanz** [comment]
It took me 10 minutes to fix it and 2 days to test it :)

**Comment by dzenanz** [comment]
appveyor build failed (could not download boost). Error on line 92:
```
  CMake Error at Boost-stamp/download-Boost.cmake:27 (message):
CUSTOMBUILD : error : downloading [c:\statismo-superbuild\Boost.vcxproj]
    'http://sourceforge.net/projects/boost/files/boost/1.56.0/boost_1_56_0.zip'
    failed
  
      status_code: 7
      status_string: "Couldn't connect to server"
      log: timeout on name lookup is not supported
      Trying 216.34.181.60...
```

**Comment by tom-albrecht** [comment]
That took forever! But I can confirm that this works and will now merge the pull request.
Can you make the same pull request for the develop branch?

**Pull request merged**